index.d.ts

```diff
@@ -0,0 +1,141 @@
+	(run: CallbackTest): void;
+	skip: CallbackRunner;
+}
+
+export function test(name: string, run: Test): void;
+export function test(run: Test): void;
+export namespace test {
+	export const before: Runner;
+	export const after: Runner;
+	export const beforeEach: Runner;
+	export const afterEach: Runner;
+
+	export const skip: typeof test;
+	export const only: typeof test;
+
+	export function serial(name: string, run: SerialTest): void;
+	export function serial(run: SerialTest): void;
+	export function cb(name: string, run: CallbackTest): void;
+	export function cb(run: CallbackTest): void;
+}
+export namespace test.serial {
+	export const before: SerialRunner;
+	export const after: SerialRunner;
+	export const beforeEach: SerialRunner;
+	export const afterEach: SerialRunner;
+
+	export const skip: typeof test.serial;
+	export const only: typeof test.serial;
+}
+export namespace test.cb {
+	export const before: CallbackRunner;
+	export const after: CallbackRunner;
+	export const beforeEach: CallbackRunner;
+	export const afterEach: CallbackRunner;
+
+	export const skip: typeof test.cb;
+	export const only: typeof test.cb;
+}
+export default test;
+
+export type ErrorValidator
+	= (new (...args: any[]) => any)
+	| RegExp
+	| string
+	| ((error: any) => boolean);
+
+export interface AssertContext {
+	/**
+	 * Passing assertion.
+	 */
+	pass(message?: string): void;
+	/**
+	 * Failing assertion.
+	 */
+	fail(message?: string): void;
+	/**
+	 * Assert that value is truthy.
+	 */
+	ok(value: any, message?: string): void;
+	/**
+	 * Assert that value is falsy.
+	 */
+	notOk(value: any, message?: string): void;
+	/**
+	 * Assert that value is true.
+	 */
+	true(value: boolean, message?: string): void;
+	/**
+	 * Assert that value is false.
+	 */
+	false(value: boolean, message?: string): void;
+	/**
+	 * Assert that value is equal to expected.
+	 */
+	is<U>(value: U, expected: U, message?: string): void;
+	/**
+	 * Assert that value is not equal to expected.
+	 */
+	not<U>(value: U, expected: U, message?: string): void;
+	/**
+	 * Assert that value is deep equal to expected.
+	 */
+	same<U>(value: U, expected: U, message?: string): void;
+	/**
+	 * Assert that value is not deep equal to expected.
+	 */
+	notSame<U>(value: U, expected: U, message?: string): void;
+	/**
+	 * Assert that function throws an error or promise rejects.
+	 * @param error Can be a constructor, regex, error message or validation function.
+	 */
+	throws(value: (() => void) | Promise<{}>, error?: ErrorValidator, message?: string);
+	/**
+	 * Assert that function doesn't throw an error or promise resolves.
+	 */
+	notThrows(value: (() => void) | Promise<{}>, message?: string);
+	/**
+	 * Assert that contents matches regex.
+	 */
+	regex(contents: string, regex: RegExp, message?: string): void;
+	/**
+	 * Assert that error is falsy.
+	 */
+	ifError(error: any, message?: string): void;
+}
+export interface TestContext extends AssertContext {
```

Comment:
Not sure why this is extending from `AssertContext`.

![screen shot 2016-02-26 at 08 57 56](https://cloud.githubusercontent.com/assets/1913805/13346506/180f27b8-dc67-11e5-95c0-d42a9ebd81b1.png)